py3status/modules/year_progress.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+"""
+Year progress.
+
+Configuration parameters:
+    progress_block: block type to show progress (default '▓')
+    remain_block: block type to show remaining progress (default '░')
+    progress_width: progress bar width (default 5)
+    cache_timeout: refresh interval for this module, default 1 hour (default 3600)
+    ctime: custom time for the end of progress, useful for appointment/birthday progress.
+
+Format placeholders:
+    {progress_bar} Progress bar
+    {ratio} Progress ratio
+    {mode} Progress mode [year|month|day|week], click the module to cycle (default 'year')
+
+
+Examples:
+```
+year_progress {
+    format = "{progress_bar} {ratio}%{mode}"
+    progress_block = '▓'
+    remain_block = '░'
+    progress_width = 5
+    ctime = ["2020-12-16 20:59:00", "%Y-%m-%d %H:%M:%S"]
+}
+```
+
+@author azzamsa <https://github.com/azzamsa>
+@license GPLv3 <https://www.gnu.org/licenses/gpl-3.0.txt>
+
+Notes:
+    Credit: Part of the code are ported from twitter.com/year_progress <https://github.com/filiph/progress_bar>
+
+SAMPLE OUTPUT
+{'full_text': '▓▓▓▓░ 87%y'}
+
+{'full_text': '▓▓▓░░ 50%m'}
+
+{'full_text': '▓▓░░░ 33%d'}
+
+{'full_text': '▓▓▓▓░ 83%w'}
+"""
+
+from datetime import datetime, tzinfo, timedelta
+
+
+class UTC(tzinfo):
+    """UTC"""
+
+    def utcoffset(self, dt):
+        return timedelta(0)
+
+    def tzname(self, dt):
+        return "UTC"
+
+    def dst(self, dt):
+        return timedelta(0)
+
+
+def compute_progress(start, end, current):
```

Comment:
(Request) Can you put all this functions inside py3status class? This way, you can print things (i.e. `self.py3.log(current_diff)` in the log/terminal. Not only that, but there are other `py3` functions that we can use sometimes for other things.

Reply:
done.